var/spack/repos/builtin/packages/py-soundfile/package.py

```diff
@@ -0,0 +1,19 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PySoundfile(PythonPackage):
+    """SoundFile is an audio library based on libsndfile, CFFI and NumPy."""
+
+    homepage = "https://github.com/bastibe/PySoundFile"
+    url      = "https://pypi.io/packages/source/S/SoundFile/SoundFile-0.10.3.post1.tar.gz"
+
+    version('0.10.3.post1', sha256='490cff42650733d1832728b937fe99fa1802896f5ef4d61bcf78cf7ebecb107b')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-cffi@1.0:', type=('build', 'run'))
+    depends_on('py-numpy', type=('build', 'run'))
```

Comment:
This dependency is optional, I would either remove it or add a variant (default to False) for it.

Reply:
According to the docs is depends on numpy. However, I see that it also depends on libsndfile, at least at run time. I have that on my system so missed it but spack needs a package for that.

Reply:
Numpy is an extra feature, when you `pip install soundfile` it doesn't use numpy unless you tell it to. And yes, it looks like it has a dependency on the `libsndfile` library.

Reply:
Well, it does not link to anything in the traditional sense. The python module will load the libsndfile library via CFFI. That feels more like a run dependency to me.

Reply:
I don't know much about CFFI...

Link dependencies add the libraries and headers via the compiler wrappers during build time.
Run dependencies add the dependencies `prefix.bin` to `PATH`

Does CFFI find libraries via PATH or something else?

Reply:
Also, I tested building without the `libsndfile` library, even removing the system library. It builds fine without it being present but it is needed at run time to be able to import `soundfile`.

Reply:
The library is found via the `find_library` function of `ctypes.util`.

Reply:
On Linux, `find_library` will use `ld`, so as long as `$LD_LIBRARY_PATH` is set by `libsndfile`, py-sound will be able to find the library.

Reply:
Would it be better just to use the default dependency type?
```
depends_on('libsndfile')
```

Reply:
Nah, the default is build/link, so it's the exact opposite of what you already have